tests/integration_tests/performance/test_snapshot_restore_performance.py

```diff
@@ -0,0 +1,360 @@
+            delta_percentage = baseline.get("delta_percentage")
+            return {
+                "target": target,
+                "delta": delta_percentage * target / 100,
+            }
+        return None
+
+
+def construct_scratch_drives():
+    """Create an array of scratch disks."""
+    scratchdisks = ["vdb", "vdc", "vdd", "vde"]
+    disk_files = [
+        drive_tools.FilesystemFile(tempfile.mktemp(), size=64)
+        for _ in scratchdisks
+    ]
+    return zip(scratchdisks, disk_files)
+
+
+def default_lambda_consumer(env_id):
+    """Create a default lambda consumer for the snapshot restore test."""
+    return st.consumer.LambdaConsumer(
+        metadata_provider=DictMetadataProvider(
+            CONFIG["measurements"],
+            SnapRestoreBaselinesProvider(env_id)
+        ),
+        func=consume_output,
+        func_kwargs={})
+
+
+def get_snap_restore_latency(
+        context,
+        vcpus,
+        mem_size,
+        nets=1,
+        blocks=1,
+        all_devices=False,
+        iterations=10):
+    """Restore snapshots with various configs to measure latency."""
+    vm_builder = context.custom['builder']
+
+    # Create a rw copy artifact.
+    rw_disk = context.disk.copy()
+    # Get ssh key from read-only artifact.
+    ssh_key = context.disk.ssh_key()
+
+    ifaces = None
+    if nets > 1:
+        ifaces = net_ifaces[:nets]
+
+    # Create a fresh microvm from artifacts.
+    vm_instance = vm_builder.build(
+        kernel=context.kernel,
+        disks=[rw_disk],
+        ssh_key=ssh_key,
+        config=context.microvm,
+        net_ifaces=ifaces,
+        use_ramdisk=True)
+    basevm = vm_instance.vm
+    response = basevm.machine_cfg.put(
+        vcpu_count=vcpus,
+        mem_size_mib=mem_size,
+        ht_enabled=False
+    )
+    assert basevm.api_session.is_status_no_content(response.status_code)
+
+    extra_disk_paths = []
+    if blocks > 1:
+        for (name, diskfile) in list(scratch_drives)[:(blocks - 1)]:
+            basevm.add_drive(name, diskfile.path, use_ramdisk=True)
+            extra_disk_paths.append(diskfile.path)
+        assert len(extra_disk_paths) > 0
+
+    if all_devices:
+        response = basevm.balloon.put(
+            amount_mib=0,
+            deflate_on_oom=True,
+            stats_polling_interval_s=1
+        )
+        assert basevm.api_session.is_status_no_content(response.status_code)
+
+        response = basevm.vsock.put(
+            vsock_id="vsock0",
+            guest_cid=3,
+            uds_path="/v.sock"
+        )
+        assert basevm.api_session.is_status_no_content(response.status_code)
+
+    basevm.start()
+
+    ssh_connection = net_tools.SSHConnection(basevm.ssh_config)
+
+    # Create a snapshot builder from a microvm.
+    snapshot_builder = SnapshotBuilder(basevm)
+    snapshot = snapshot_builder.create(
+        [rw_disk.local_path()] + extra_disk_paths,
+        ssh_key,
+        SnapshotType.FULL,
```

Comment:
nit: since this test is only for restoring full snapshots, would make sense to name the file or the function accordingly?

Reply:
Done.